### PR TITLE
login route

### PR DESCRIPTION
**Pull request by jlurgo**
- adds /login route to react-router
  - now dashboard is not visible on the background
- adds login failed error to login dialog
- adds requireLogin middleware to all dashboard routes with server side redirecting to login
- some cleanup

### CODE REVIEW

File: src/client/components/user-context.tsx [hunk 1]
```diff
@@ -20,29 +24,48 @@ export const UserContextProvider = ({children}) => {
   };
 
   useEffect(() => {
-      // refresh cookie
-      fetchJson(`/refresh`, (err, res) => {
```
Comment (chfritz) [inline src/client/components/user-context.tsx:24]: Did this not work for your needs? What was missing?

File: src/server/auth.ts [hunk 2]
```diff
@@ -48,12 +47,19 @@ export const writeAccounts = async (accounts: Account[]): Promise<void> => {
...
-        res.redirect('/login');
-    } else {
-        next();
-    }
+  if (req.session && req.session.user) {
+    next();
+    return;
+  }
+  log.debug('not logged in', req.url);
+  if (req.xhr || req.headers.accept.indexOf('json') > -1) {
```
Comment (chfritz) [inline src/server/auth.ts:58]: This seems unnecessarily complicated. We control the other side, so guessing whether the client accepts json is not required, we can just enforce it, no? My sense is that we can just always use `res.direct` now that we have a proper login route.

**Review by chfritz — COMMENTED** [review]
Looks good! Some comments inline plus:
- The logout button works (logs the user out) but doesn't yet redirect 
- When logged in the `/` route doesn't work.

File: src/client/components/user-context.tsx (hunk 1 above)
Comment (jlurgo) [reply src/client/components/user-context.tsx:24]: The problem here was with res.redirect() from the server side, it ended up with an error parsing html as json.

**Review by chfritz — APPROVED** [review]
Works great!

**Pull request merged**